### PR TITLE
Recently, when we looked at the code, we discovered that there is a p…

### PR DESCRIPTION
**Pull request by kim-chunghee**
…roblem when a user creates a transaction with two or more payloads. This problem only impacts the js wallet, though. Recently, a contributor slipped the bug into the code. Please review the changes and merge them as soon as you can to solve the issue.

### CODE REVIEW

**Comment by xiangwei22** [comment]
it would be beneficial if you could describe the problem and the solution.

**Comment by kim-chunghee** [comment]
In cases where there are two or more payloads, the code will generate identical sender rings with different permutations. These permutations will reveal the real sender. Just the JS wallet contains this error.

**Comment by xiangwei22** [comment]
ok. i see. i am the one in charge of pandora cash a fork of pandora pay. do you think we should update our builds with your fixes?

**Comment by kim-chunghee** [comment]
Better if it happens sooner.

**Comment by xiangwei22** [comment]
ok. thank you.

**Comment by ibudisteanu** [comment]
WOOOW. So many commits and files changed. It will take me some time to review all your changes.

**Comment by ibudisteanu** [comment]
There are so many changes... you should have used different branches or make pull requests. It is really hard to follow the changes.

**Comment by ibudisteanu** [comment]
Why did you rename some of the files? Was it necessary?

**Comment by kim-chunghee** [comment]
Your remarks were noted. For the inconvenience, we sincerely apologize. We'll provide more clarification to our commits when we open a new pull request.

**Comment by ibudisteanu** [comment]
Sorry for delay. I have been really busy at home. Your pull request is so big. It is really hard to follow your changes.

**Comment by ibudisteanu** [comment]
I really appreciate your work but you should split the changes into multiple branches / or pull requests

**Comment by ibudisteanu** [comment]
Thanks for your hard work!
